src/components/Pickers/Suggestion/SuggestionController.ts

```diff
@@ -0,0 +1,85 @@
+export interface ISuggestionModel {
+  item: any;
+  isSelected: boolean;
+}
+
+export class SuggestionController {
+  public currentIndex: number;
+  public currentSuggestion: ISuggestionModel;
+  private suggestions: ISuggestionModel[];
+  constructor() {
+      this.suggestions = [];
+      this.currentIndex = -1;
+  }
+
+  public updateSuggestions(newSuggestions: any[]) {
+    if (newSuggestions && newSuggestions.length > 0) {
+      this.suggestions = this._convertSuggestionsToSuggestionItems(newSuggestions);
+      this.currentIndex = 0;
+      this.suggestions[0].isSelected = true;
+      this.currentSuggestion = this.suggestions[0];
+    } else {
+      this.suggestions = [];
+      this.currentIndex = -1;
+      this.currentSuggestion = undefined;
+    }
+  }
+
+  public nextSuggestion(): boolean {
+    if (this.currentIndex < (this.suggestions.length - 1)) {
+      this.setSelectedSuggestion(this.currentIndex + 1);
+      return true;
+    } else if (this.currentIndex === (this.suggestions.length - 1)) {
+      this.setSelectedSuggestion(0);
+      return true;
+    }
+  }
+
+  public previousSuggestion(): boolean {
+    if (this.currentIndex > 0) {
+      this.setSelectedSuggestion(this.currentIndex - 1);
+      return true;
+    } else if (this.currentIndex === 0) {
+      this.setSelectedSuggestion(this.suggestions.length - 1);
+      return true;
+    }
+  }
+
+  public getSuggestions(): ISuggestionModel[] {
+    return this.suggestions;
+  }
+
+  public getCurrentItem(): ISuggestionModel {
+    return this.currentSuggestion;
+  }
+
+  public getSuggestionAtIndex(index: number): ISuggestionModel {
+    return this.suggestions[index];
+  }
+
+  public hasSelectedSuggestion(): boolean {
+    return this.currentSuggestion ? true : false;
```

Comment:
Probably, but I tend to like things a little more explicit. For me it's easier to understand :/.